### PR TITLE
Adding Badge and NumericBadge components

### PR DESCRIPTION
**Pull request by ashleemboyer**
## "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [x] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## Relevant links

- [Badge preview link](https://dev-portal-git-ambadd-badge-component-hashicorp.vercel.app/swingset/components/badge) 🔎
- [NumericBadge preview link](https://dev-portal-git-ambadd-badge-component-hashicorp.vercel.app/swingset/components/numericbadge) 🔎
- [Asana task](https://app.asana.com/0/1201979821989355/1202022787106822/f) 🎟️

## What

- Adds `Badge`
- Adds `NumericBadge`, which renders a `Badge` with some small CSS differences
- Adds a few more icons to `SwingsetTestIcon`

## Why

- `Badge` is used in some Learn pages.
- Not sure if `NumericBadge` is in designs yet, but was easy enough to add along with `Badge`.

## How

- Mainly looked at the styles here: https://design-system-components-hashicorp.vercel.app/components/badge
- `NumericBadge` uses `Badge` since there are only a few differences between the two in terms of style and props options. TypeScript will help us with the latter, mainly with the the limited `color` options.

## Testing

- [ ] Explore [the `Badge` swingset page](https://dev-portal-git-ambadd-badge-component-hashicorp.vercel.app/swingset/components/badge) and see if it behaves how you expect
- [ ] Explore [the `NumericBadge` swingset page](https://dev-portal-git-ambadd-badge-component-hashicorp.vercel.app/swingset/components/numericbadge) and see if it behaves how you expect

## Anything else?

Nope!

### CODE REVIEW

File: src/components/count-badge/index.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import classNames from 'classnames'
+import Badge from 'components/badge'
+import { CountBadgeProps } from './types'
+import s from './badge-count.module.css'
+
+const CountBadge = ({
```
Comment (BRKalow) [inline src/components/count-badge/index.tsx:6]: Bike-shedding alert: do you think `NumericBadge` would be a better name? The Count prefix felt confusing when I saw there were semver versions rendered with this component.

File: src/components/badge/index.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    throw new Error(
+      'Icon-only `Badge`s require an accessible label. Either provide the `text` prop or one of: `ariaLabel`, `ariaLabelledBy`, `ariaDescribedBy`.'
+    )
+  }
+
+  if (ariaLabel && ariaLabelledBy) {
+    throw new Error(
+      '`Badge` does not accept both `ariaLabel` and `ariaLabelledBy`. Only provide one or the other.'
+    )
+  }
```
Comment (BRKalow) [inline src/components/badge/index.tsx:46]: 👍

**Review by BRKalow — APPROVED** [review]
The implementation looks good to me!

**Review by kendallstrautman — APPROVED** [review]
🚀

File: src/components/count-badge/index.tsx (hunk 1 above)
Comment (ashleemboyer) [reply src/components/count-badge/index.tsx:6]: I agree, `NumericBadge` is a little clearer! I'll make this change. 😊

**Review by zchsh — APPROVED** [review]
Looks great!

Very much a non-blocking nit-picky take-it-or-leave-it suggestion, but regardless I'll submit the idea: we could name this `BadgeNumeric` so it shows up in file lists and in Swingset next to `Badge`... but also `BadgeNumeric` certain feels less natural to say than `NumericBadge` so  🤷‍♂️

**Pull request merged**